internal/controllers/application.go

```diff
@@ -299,6 +311,10 @@ func (r *ApplicationReconciler) setApplicationDefaults(application *skiperatorv1
 		}
 	}
 
+	//We try to feed the access policy with port values dynamically,
+	//if unsuccessfull we just don't set ports, and rely on podselectors
+	r.UpdateAccessPolicy(ctx, application)
+
```

Comment:
The only reasons that would happen is a typo (doesn't exist) or if the kubernetes api is unresponsive for some reason. I was just thinking that sometimes you might want to communicate with a different app, but it might not always be up, and your application will still work fine without it, so this shouldn't stop your app from running. I added the Invalid rule condition for the purpose of alerting if its misconfigured, so we (or teams) could create some default grafana alerts for example. I'm not sure if this is the correct approach as you say, because I'm sure some users will be lazy and just leave bad config